marian-nmt/marian-dev
# Improved handling of SIGTERM

### Description
- moved signal handling out of scheduler code (was already global there, but still in the files)
- pay attention to the signal in the threads that prepare batches in the background, which can take quite a while and prevented Marian from shutting down quickly after receiving SIGTERM in some circumstances

### How to test

start training, then send SIGTERM to process, it should save the current state of affairs and exit

### Checklist
